layout/new.go

```diff
@@ -58,7 +59,7 @@ func NewImage(path string, ops ...ImageOption) (*Image, error) {
 		ri.createdAt = imageOpts.createdAt
 	}
 
-	if imageOpts.mediaTypes == imgutil.MissingTypes {
+	if imageOpts.mediaTypes == imgutil.MissingTypes && !hasBaseImage {
```

Comment:
Following my previous comment. 
The test is providing a base image and the option to override the media type, so that's why the it is never getting into this line and the media-type is not overrided

Reply:
Maybe something like this should do the trick: ?
```go
	if imageOpts.mediaTypes != imgutil.MissingTypes {
		ri.requestedMediaTypes = imageOpts.mediaTypes
	} 
```

Reply:
Yeah. The idea is that if `layout.WithMediaTypes` is omitted, the `imageOpts.mediaTypes` will be `imgutil.MissingTypes`. In the case where there was a base image (always true in sparse, not always true in layout) - we should default to matching the base image.